src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,23 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import jakarta.persistence.EntityNotFoundException;
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao = new CountryDaoImpl(HibernateUtil.getSessionFactory());
```

Comment:
Direct instantiation of the DAO implementation within the service layer is not recommended. Instead, consider using dependency injection to manage your DAO instances. This will make your code more modular, easier to test, and aligns with the principles of inversion of control.